Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -22,47 +19,63 @@ class SpriteRepository {
     
     var sessionConfiguration: URLSessionConfiguration = URLSessionConfiguration.default {
         didSet {
-            imageDownloader = ImageDownloader(sessionConfiguration: sessionConfiguration)
+            requestDownloader = RequestDownloader(sessionConfiguration: sessionConfiguration)
         }
     }
     
+    required init(withDownloader downloader: ReentrantRequestDownloader = RequestDownloader(),
+                  requestCache: BimodalURLCache = URLDataCache(),
+                  derivedCache: BimodalImageCache = ImageCache()) {
+        requestDownloader = downloader
+        self.requestCache = requestCache
+        self.derivedCache = derivedCache
+    }
+    
+    func urlRequest(with url: URL) -> URLRequest {
+        var request = URLRequest(url: url)
+        request.allHTTPHeaderFields = self.headers
+        request.cachePolicy = .reloadIgnoringCacheData
```

Comment:
Because we check whether the response is cached in the `URLCache` before the downloading. Only when it's not cached, we send the urlRequest to download the images and metadata to the `URLCache`. And the `URLCache` won't discard the response when the cachePolicy is `. reloadIgnoringCacheData `.

Reply:
Does this mean we don’t need to explicit check beforehand whether something is in the cache if we use a different policy? Relying on the cache to do that check automatically could reduce the potential for errors on our side, in case in the future there’s a code path in which we neglect to check first. On the other hand, the default policy is dependent on the correct caching headers being sent by the server; I’m unsure if relying on the current headers would result in the same behavior as that explicit check.

Reply:
Yes. If we choose other policy such as the default `.useProtocolCachePolicy`, it would automatically check the existence of the cache when downloading without the manually check. But the issue is that if we don't use the manually check, they'll all call the `RequestDownloader` to download and are added to the download queue. But if we manually check the existence of the URLCaChe, then it would skip the downloading process and return the image and info more quickly directly from the cache. It follows the same logic as the `ImageRepository` for handling image as before.